doc/getting-started/gsg_quick_setup.sh

```diff
@@ -0,0 +1,207 @@
+
+# Compare with current clear linux and skip upgrade sos if get the same version.
+if [[ $(echo -e `swupd info` | cut -d' ' -f3) -eq $target_version ]]; then
+    echo "Specified to the same clear linux version, continue to setup sos env."
+else
+    echo "Upgrade clear linux version to : $target_version ..."
+    swupd verify --fix --picky -m $target_version
+fi
+
+# Do the setups if previous process succeed.
+if [[ $? -eq 0 ]]; then
+    echo "Add service-os kernel-iot-lts2018 bundle"
+    swupd bundle-add service-os kernel-iot-lts2018
+
+    echo "mount /dev/sda1 to /mnt"
+    mount /dev/sda1 /mnt
+
+    echo "Add /mnt/EFI/acrn folder"
+    mkdir -p /mnt/EFI/acrn
+    echo "Copy /usr/share/acrn/samples/nuc/acrn.conf /mnt/loader/entries/"
+    cp -r /usr/share/acrn/samples/nuc/acrn.conf /mnt/loader/entries/
+    echo "Copy acrn.efi to /mnt/EFI/acrn"
+    cp -r /usr/lib/acrn/acrn.efi /mnt/EFI/acrn/
+
+    echo "Check ACRN efi boot event"
+    check_acrn_bootefi=`efibootmgr | grep ACRN`
+    if [[ "$check_arcn_bootefi" -ge "ACRN" ]]; then
+        echo "Clean all ACRN efi boot event"
+        efibootmgr | grep ACRN | cut -d'*' -f1 | cut -d't' -f2 | xargs -i efibootmgr -b {} -B
+    fi
+    echo "Check linux bootloader event"
+    number=$(expr `efibootmgr | grep 'Linux bootloader' | wc -l` - 1)
+    if [[ $number -ge 1 ]]; then
+        echo "Clean all Linux bootloader event"
+        efibootmgr | grep 'Linux bootloader' | cut -d'*' -f1 | cut -d't' -f2 | head -n$number | xargs -i efibootmgr -b {} -B
+    fi
+
+    echo "Add new ACRN efi boot event"
+    efibootmgr -c -l "\EFI\acrn\acrn.efi" -d /dev/sda -p 1 -L "ACRN"
+
+    echo "Create loader.conf"
+    rm /mnt/loader/loader.conf && touch /mnt/loader/loader.conf
+    echo "Add default boot wait time"
+    echo "timeout 5" > /mnt/loader/loader.conf
+    echo "Add default boot to ACRN"
+    echo "default acrn" >> /mnt/loader/loader.conf
+
+    echo "Get latest sos file from EFI org.clearlinux: "
+    new_kernel=`ls /mnt/EFI/org.clearlinux/*sos* -tl | grep kernel | head -n1 | awk -F'/' '{print $5}'`
+    echo "Get current sos kernel from acrn.conf file: "
+    cur_kernel=`cat /mnt/loader/entries/acrn.conf | sed -n 2p | cut -d'/' -f4`
+
+    echo "Replace root uuid from acrn.conf"
+    sed -i "s/<UUID of rootfs partition>/`blkid -s PARTUUID -o value /dev/sda3`/g" /mnt/loader/entries/acrn.conf
+
+    if [[ "$cur_kernel" == "$new_kernel" ]]; then
+        echo "No need to replace kernel conf info"
+    else
+        echo "Replace with new sos kernel in acrn.conf"
+        sed -i "s/$cur_kernel/$new_kernel/g" /mnt/loader/entries/acrn.conf
+    fi
+
+    echo "Sos process done!"
+    echo " ***** Please reboot device to test on new sos. *****"
+else
+    echo -e "Fail to upgrade sos $target_clearlinux from mirror url."
+fi
+
+echo "Init swupd mirror & proxy"
+swupd mirror -u
+unset https_proxy
+umount /mnt
+}
+
+function upgrade_uos()
+{
+# Add proxy if you need to download uos img automatically.
```

Comment:
You never download the UOS image automatically in the script, or did I miss it?

If true, then no need to modify any proxy setting

Reply:
I have update in this part. Now it is also support to download uos image from official site with proxy settings.